src/Reflection/ReflectionClass.php

```diff
@@ -984,12 +1015,60 @@ static function (ReflectionClass $trait) : string {
      */
     public function getTraitAliases() : array
     {
+        $this->parseTraitUsages();
+
+        /** @return array<string, string> */
+        return $this->cachedTraitAliases;
+    }
+
+    /**
+     * Return a list of the precedences used when importing traits for this class.
+     * The returned array is in key/value pair in this format:.
+     *
+     *   'Class::method' => 'Class::method'
+     *
+     * @return array<string, string>
+     *
+     * @example
+     * // When reflecting a class such as:
+     * class Foo
+     * {
+     *     use MyTrait, MyTrait2 {
+     *         MyTrait2::foo insteadof MyTrait1;
+     *     }
+     * }
+     * // This method would return
+     * //   ['MyTrait1::foo' => 'MyTrait2::foo']
+     */
+    public function getTraitPrecedences() : array
```

Comment:
we should probably **not** expose this API. It is complex enough to require complex examples, and I can't see a use-case for it where downstream would care about trait precedence?